rosbag2_transport/test/rosbag2_transport/test_record_all.cpp

```diff
@@ -104,6 +104,9 @@ TEST_F(RecordIntegrationTestFixture, published_messages_from_multiple_services_a
 
   start_async_spin(recorder);
 
+  // Await the recorder to start spinning
+  std::this_thread::sleep_for(300ms);
+
```

Comment:
@Barry-Xu-2018 Could you please explain why do you need this sleep in test?
Need to avoid sleep in tests - it is a red flag signaling that the test is not deterministic and tend to be flaky.

Reply:
@MichaelOrlov 

After checking the code, I think this code is redundant. I will delete this code.

Reply:
Done. Commit is b921d66.